### PR TITLE
Resolve "Additional Enhancements for LOINC terminology"

### PR DESCRIPTION
**Pull request by dmuylwyk**
- See #1451 / https://github.com/jamesagnew/hapi-fhir/pull/1485/commits/9b1af6b20783abfb99df9de0ccaed69dab7250ec
- See #1453 / https://github.com/jamesagnew/hapi-fhir/commit/b05c5f3f711b078515c026e9cd09e0abb1f9e733

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java [hunk 1]
```diff
@@ -925,6 +874,106 @@ private Boolean expandValueSetHandleIncludeOrExclude(IValueSetConceptAccumulator
...
+
+		if (isBlank(nextFilter.getValue()) || nextFilter.getOp() == null || isBlank(nextFilter.getProperty())) {
+			throw new InvalidRequestException("Invalid filter, must have fields populated: property op value");
+		}
+
+		if (nextFilter.getProperty().equals("display:exact") || nextFilter.getProperty().equals("display")) {
+			handleDisplayFilter(theQb, theBool, nextFilter);
+		} else if (nextFilter.getProperty().equals("concept") || nextFilter.getProperty().equals("code")) {
+			handleConceptAndCodeFilter(theSystem, theQb, theBool, nextFilter);
+		} else if (nextFilter.getProperty().equals("copyright")) {
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java]: You should probably add an additional check here that the include being processed has the loinc codesystem URL... We wouldn't want to allow this particular property to work for any other codesystem

Comment (dmuylwyk) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java]: Done.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java [hunk 2]
```diff
@@ -925,6 +854,195 @@ private Boolean expandValueSetHandleIncludeOrExclude(IValueSetConceptAccumulator
...
+		if (theFilter.getOp() == ValueSet.FilterOperator.ISA) {
+			ourLog.info(" * Filtering on codes with a parent of {}/{}/{}", code.getId(), code.getCode(), code.getDisplay());
+			theBool.must(theQb.keyword().onField("myParentPids").matching("" + code.getId()).createQuery());
+		} else {
+			throw new InvalidRequestException("Don't know how to handle op=" + theFilter.getOp() + " on property " + theFilter.getProperty());
+		}
+	}
+
+	private void handleFilterLoincParentChild(String theSystem, QueryBuilder theQb, BooleanJunction<?> theBool, ValueSet.ConceptSetFilterComponent theFilter) {
+		if (isNotCodeSystemLoinc(theSystem)) {
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java]: I'd move this check out of this method and into the caller.. If another codesystem ever has its own parent/child properties it'll be confusing to see this method get called but not do anything

Comment (dmuylwyk) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java]: Done.

**Pull request merged**